airbyte-cdk/bulk/core/extract/src/main/kotlin/io/airbyte/cdk/read/StateManager.kt

```diff
@@ -91,48 +91,53 @@ class StateManager(
     fun checkpoint(): List<AirbyteStateMessage> =
         listOfNotNull(global?.checkpoint()) + nonGlobal.mapNotNull { it.value.checkpoint() }
 
+    private sealed interface StateForCheckpoint {
+        val opaqueStateValue: OpaqueStateValue?
+        val numRecords: Long
+    }
+
+    private data class Fresh(
```

Comment:
Can you add a comment for the purpose behind fresh and stale

Reply:
done